client/client.go

```diff
@@ -0,0 +1,1375 @@
+	MaxPerChunk         int32 `json:"max_per_chunk,omitempty"`
+	MaxParallelRequests int32 `json:"max_parallel_requests,omitempty"`
+}
+
+type ClientWriteOptions struct {
+	AuthorizationModelId *string             `json:"authorization_model_id,omitempty"`
+	Transaction          *TransactionOptions `json:"transaction_options,omitempty"`
+}
+
+type ClientWriteStatus string
+
+// List of ClientWriteStatus
+const (
+	SUCCESS ClientWriteStatus = "CLIENT_WRITE_STATUS_SUCCESS"
+	FAILURE ClientWriteStatus = "CLIENT_WRITE_STATUS_FAILURE"
+)
+
+//var allowedTupleOperationEnumValues = []ClientWriteStatus{
+//	SUCCESS,
+//	FAILURE,
+//}
+
+type ClientWriteSingleResponse struct {
+	TupleKey ClientTupleKey
+	Status   ClientWriteStatus
+	Response *_nethttp.Response
+	Error    error
+}
+
+type ClientWriteResponse struct {
+	Writes  []ClientWriteSingleResponse
+	Deletes []ClientWriteSingleResponse
+}
+
+func (client *OpenFgaClient) Write(ctx _context.Context) SdkClientWriteRequest {
+	return SdkClientWriteRequest{
+		Client: *client,
+		ctx:    ctx,
+	}
+}
+
+func (request SdkClientWriteRequest) Options(options ClientWriteOptions) SdkClientWriteRequest {
+	request.options = &options
+	return request
+}
+
+func (request SdkClientWriteRequest) Body(body ClientWriteRequest) SdkClientWriteRequest {
+	request.body = &body
+	return request
+}
+
+func (request SdkClientWriteRequest) Execute() (ClientWriteResponse, error) {
+	return request.Client.WriteExecute(request)
+}
+
+func (client *OpenFgaClient) WriteExecute(request SdkClientWriteRequest) (ClientWriteResponse, error) {
+	var maxPerChunk = int32(1) // 1 has to be the default otherwise the chunks will be sent in transactions
+	if request.options != nil && request.options.Transaction != nil {
+		maxPerChunk = request.options.Transaction.MaxPerChunk
+	}
+	var maxParallelReqs = DEFAULT_MAX_METHOD_PARALLEL_REQS
+	if request.options != nil && request.options.Transaction != nil {
+		maxParallelReqs = request.options.Transaction.MaxParallelRequests
+	}
+
+	if request.options == nil || request.options.Transaction == nil || !request.options.Transaction.Disable {
+		writeRequest := openfga.WriteRequest{
+			AuthorizationModelId: client.getAuthorizationModelId(request.options.AuthorizationModelId),
+		}
+		if request.body.Writes != nil && len(*request.body.Writes) > 0 {
+			writes := openfga.TupleKeys{}
+			for index := 0; index < len(*request.body.Writes); index++ {
+				writes.TupleKeys = append(writes.TupleKeys, (*request.body.Writes)[index].ToTupleKey())
+			}
+			writeRequest.Writes = &writes
+		}
+		if request.body.Deletes != nil && len(*request.body.Deletes) > 0 {
+			deletes := openfga.TupleKeys{}
+			for index := 0; index < len(*request.body.Deletes); index++ {
+				deletes.TupleKeys = append(deletes.TupleKeys, (*request.body.Deletes)[index].ToTupleKey())
+			}
+			writeRequest.Deletes = &deletes
+		}
+		_, httpResponse, err := client.OpenFgaApi.Write(request.ctx).Body(writeRequest).Execute()
```

Comment:
if `len(*request.body.Writes) = 100000000`, won't this fail? should we not chunk requests?

Reply:
In transaction mode, we shouldn't chunk - we send the data as is to the server. Is is up to the server to accept or reject that request.